zstacklib/zstacklib/utils/file_metadata_handler.py

```
@@ -0,0 +1,188 @@
+        "architecture": architecture,
+        "schemaVersion": schema_version,
+    }, ensure_ascii=False, separators=(',', ':'))
+    try:
+        fd = os.open(summary_tmp, os.O_WRONLY | os.O_CREAT | os.O_TRUNC, 0o600)
+        with os.fdopen(fd, 'w') as f:
+            f.write(summary_json)
+            f.flush()
+            os.fsync(f.fileno())
+        os.rename(summary_tmp, summary_path)
+        _fsync_directory(summary_path)
+    except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)